testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/exception/ClosedResponseHandlingTest.java

```diff
@@ -0,0 +1,160 @@
+import org.jboss.resteasy.plugins.server.servlet.ResteasyContextParameters;
+import org.jboss.resteasy.tracing.RESTEasyTracingLogger;
+import org.jboss.resteasy.tracing.api.RESTEasyTracing;
+import org.jboss.resteasy.tracing.api.RESTEasyTracingLevel;
+import org.jboss.resteasy.utils.PermissionUtil;
+import org.jboss.resteasy.utils.PortProviderUtil;
+import org.jboss.resteasy.utils.TestUtil;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static javax.ws.rs.core.Response.Status.NOT_ACCEPTABLE;
+import static javax.ws.rs.core.Response.Status.UNSUPPORTED_MEDIA_TYPE;
+
+/**
+ * @tpSubChapter Resteasy-client
+ * @tpChapter Integration tests
+ * @tpSince RESTEasy 4.0.0.CR1
+ * @tpTestCaseDetails Regression test for RESTEASY-1142
+ * @author <a href="ron.sigal@jboss.com">Ron Sigal</a>
+ * @author <a href="jonas.zeiger@talpidae.net">Jonas Zeiger</a>
+ * @version $Revision: 1.0 $
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+public class ClosedResponseHandlingTest {
+
+   @Deployment
+   public static Archive<?> deploy() {
+       WebArchive war = TestUtil.prepareArchive(ClosedResponseHandlingTest.class.getSimpleName());
+       war.addClass(ClosedResponseHandlingTest.class);
+       war.addClass(PortProviderUtil.class);
+       war.addAsManifestResource(PermissionUtil.createPermissionsXmlAsset(
+             new ReflectPermission("suppressAccessChecks")
+       ), "permissions.xml");
+
+       Map<String, String> params = new HashMap<>();
+       params.put(ResteasyContextParameters.RESTEASY_TRACING_TYPE, ResteasyContextParameters.RESTEASY_TRACING_TYPE_ALL);
+       params.put(ResteasyContextParameters.RESTEASY_TRACING_THRESHOLD, ResteasyContextParameters.RESTEASY_TRACING_LEVEL_VERBOSE);
+
+       return TestUtil.finishContainerPrepare(war, params, TestResource.class,
+             PleaseMapExceptionMapper.class,
+             TestEnableVerboseTracingRequestFilter.class);
+    }
+
+   /**
+    * @tpTestDetails Request is sent to an endpoint that issues a Resteasy client request triggering a 404 error.
+    * @tpPassCrit A NotAcceptableException is returned
+    * @tpSince RESTEasy 4.0.0.CR1
+    */
+   @Test(expected = NotAcceptableException.class)
+   public void testNotAcceptable() {
+      new ResteasyClientBuilderImpl().build().target(generateURL("/testNotAcceptable")).request().get(String.class);
+   }
+
+   /**
+    * @tpTestDetails Request is sent to an exception mapping and tracing endpoint that issues a Resteasy client request triggering a 404 error.
+    * @tpPassCrit A NotSupportedException is returned
+    * @tpSince RESTEasy 4.0.0.CR1
+    */
+   @Test(expected = NotSupportedException.class)
+   public void testNotSupportedTraced() {
+      new ResteasyClientBuilderImpl().build().target(generateURL("/testNotSupportedTraced")).request().get(String.class);
+   }
+
+   @Path("")
+   public static class TestResource {
```

Comment:
Hi @lyind.

The convention in resteasy-integration-tests is to put supporting classes like TestResource, PleaseMapException, and PleaseMapExceptionMapper in the related .../resource directory, org.jboss.resteasy.test.exception.resource in this case. The names would be related to the test class name; e.g., ClosedResponseHandlingResource, ClosedResponseHandlingPleaseMapException, etc.

Reply:
Layout re-factored in dd2ebd4f16c8c6994a0fcfe6c877d1a840f1891c.